src/Microsoft.Azure.WebJobs.Host/Tables/TableAttributeBindingProvider.cs

```diff
@@ -68,32 +62,156 @@ private TableAttributeBindingProvider(INameResolver nameResolver, IStorageAccoun
 
             return new ResolvedTableAttribute(attrResolved, client);
         }
-        
+
         public static IBindingProvider Build(INameResolver nameResolver, IConverterManager converterManager, IStorageAccountProvider accountProvider, IExtensionRegistry extensions)
         {
             var original = new TableAttributeBindingProvider(nameResolver, accountProvider, extensions);
 
             converterManager.AddConverter<JObject, ITableEntity, TableAttribute>(original.JObjectToTableEntityConverterFunc);
+            converterManager.AddConverterBuilder<object, ITableEntity, TableAttribute>(original.BuildITEConverter);
+
+            // IStorageTable --> IQueryable<ITableEntity>
+            converterManager.AddConverterBuilder<IStorageTable, TableQueryableOpenType, TableAttribute>(
+                original.QueryableConverter);
 
             var bindingFactory = new BindingFactory(nameResolver, converterManager);
-            var bindAsyncCollector = bindingFactory.BindToAsyncCollector<TableAttribute, ITableEntity>(original.BuildFromTableAttribute, null, original.CollectAttributeInfo);
 
-            var bindToJobject = bindingFactory.BindToExactAsyncType<TableAttribute, JObject>(original.BuildJObject, null, original.CollectAttributeInfo);
-            var bindToJArray = bindingFactory.BindToExactAsyncType<TableAttribute, JArray>(original.BuildJArray, null, original.CollectAttributeInfo);
+            var bindToExactCloudTable = bindingFactory.BindToGeneralAsyncType<TableAttribute, CloudTable>(
+                original.BindToCloudTable,
+                original.ToParameterDescriptorForCollector,
+                original.CollectAttributeInfo);
+
+            // Includes converter manager, which provides access to IQueryable<ITableEntity>
+            var bindToExactTestCloudTable = bindingFactory.BindToGeneralAsyncType<TableAttribute, IStorageTable>(
+                original.BindToTestCloudTable,
+                original.ToParameterDescriptorForCollector,
+                original.CollectAttributeInfo);
+
+            var bindAsyncCollector = bindingFactory.BindToAsyncCollector<TableAttribute, ITableEntity>(
+                original.BuildFromTableAttribute,
+                null,
+                original.CollectAttributeInfo);
+
+            var bindToJobject = bindingFactory.BindToExactAsyncType<TableAttribute, JObject>(
+                original.BuildJObject,
+                null,
+                original.CollectAttributeInfo);
```

Comment:
I just looked up CollectAttributeInfo to see what it does. It looks like ultimately, it is creating the client and storing it in the ResolvedTableAttribute so it can later be extracted. Our other bindings create their clients lazily and store them in ConcurrentDictionaries to be pulled out based on connection strings (so one client per conn str). Couldn't you do the same thing here and remove the concept of the 'postResolveHook' completely?